### DISCUSSION TITLE
Can't connect to server behind velocity proxy

### DISCUSSION THREAD

**Discussion by irishgreencitrus** [valence-rs/valence#368] [Q&A] [answered]
I've enabled the velocity proxy `ConnectionMode` as well as setting the compression threshold to `None` in `CoreSettings`, however every time I try to join through velocity, I get kicked.

I get this error message through valence before the kicking occurs.
```rs
WARN schedule{name=Outer}:system{name="bevy_app::CoreSchedule::outer_loop"}:schedule{name=Main}:multithreaded executor:system{name="valence_client::flush_packets"}: valence_client: Failed to flush packet queue for client 0v12: reached configured outgoing limit of 8388608 bytes.
```

Valence commit: 61f2279831920dcbc78566f68e16965ae9f07572

**Comment by dyc3** [comment]
In your velocity config, try:
```
player-info-forwarding-mode = "modern"
```

**Reply by irishgreencitrus** [reply]
Sorry, I forgot to mention that setting. It is already set as you've supplied. I've also tried `legacy` with `ConnectionMode::offline` but it doesn't seem to work. Also, I wish to have other servers running behind velocity, so using any setting other than `modern` is unideal as it requires changing the settings for my paper servers.

**Reply by dyc3** [reply]
I'm unable to reproduce this. Are you getting disconnected immediately after trying to connect? What version of minecraft are you using?

**Reply by irishgreencitrus** [reply]
I'm using 1.20.1, with [this version of velocity](https://api.papermc.io/v2/projects/velocity/versions/3.2.0-SNAPSHOT/builds/259/downloads/velocity-3.2.0-SNAPSHOT-259.jar). 
Valence kicks me (or disconnects me or something) almost immediately, but I do see the world load in, and then Velocity gets confused and sends me back to the server again, kicking me and the cycle repeats.
I have a feeling its something to do with packet compression, (similar to #262, maybe?)

**Reply by dyc3** [reply, marked as answer]
Try increasing the outgoing byte limit in `NetworkSettings`

```rust
outgoing_byte_limit: 8388608 * 4,
```

**Reply by dyc3** [reply]
@irishgreencitrus just following up, did that work?

**Reply by irishgreencitrus** [reply]
@dyc3 I've just tested it now and I do seem to stay connected, however the server seems to take quite a few seconds to respond to an event. On the `parkour.rs` example, it takes a good second or two before the score is incremented. Its technically solved my problem, so I'll mark it as an answer but they're might be a deeper issue with the way the servers are set up.

**Reply by irishgreencitrus** [reply]
Ignore me, if I re-enable velocity's compression options it works as intended. Thanks so much :D
